### PR TITLE
Move freedBytes inside group reclaiming method

### PR DESCRIPTION
**Pull request by tanjialiang**
Move the freedBytes condition inside reclaiming method in shared arbitrator. At the same time rename targetBytes to requestBytes for future PR.

### CODE REVIEW

File: velox/common/memory/SharedArbitrator.cpp [hunk 1]
```diff
@@ -283,45 +283,44 @@ uint64_t SharedArbitrator::decrementFreeCapacityLocked(
...
-  }
+
+  uint64_t fastReclaimTargetBytes =
+      std::max(memoryPoolTransferCapacity_, requestBytes);
+
   std::lock_guard<std::shared_mutex> exclusiveLock(arbitrationLock_);
   getCandidateStats(&op);
   uint64_t freedBytes =
-      reclaimFreeMemoryFromCandidates(&op, targetBytes, false);
+      reclaimFreeMemoryFromCandidates(&op, fastReclaimTargetBytes, false);
```
Comment (xiaoxmeng) [inline velox/common/memory/SharedArbitrator.cpp:310]: std::max(memoryPoolTransferCapacity_, requestBytes)? directly use it here?

File: velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp [hunk 2]
```diff
@@ -475,32 +475,34 @@ TEST_F(DateTimeFunctionsTest, yearDate) {
 }
 
 TEST_F(DateTimeFunctionsTest, yearTimestampWithTimezone) {
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp:477]: Why we need this change? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang thanks!

File: velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp (hunk 2 above)
Comment (tanjialiang) [reply velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp:477]: We don't need this change. Some unrelated changes slips through. I'll delete them

**Pull request closed without merging**